src/metabase/driver/postgres.clj

```diff
@@ -298,6 +302,73 @@
   [driver [_ arg]]
   (sql.qp/->honeysql driver [:percentile arg 0.5]))
 
+(defn- datetime-diff-helper [x y unit]
+  (case unit
+    (:year :day)
+    (hx/cast
+     :integer
+     (hsql/call
+      :date_part
```

Comment:
Would the `extract-integer` function we use for the `sql.qp/date` implementations in this namespace work here? If so, it seems like you could either use that or leverage the existing date extraction implementations we have for `sql.qp/date` instead of reinventing the wheel and duplicating a lot of that functionality